### ISSUE TITLE
[v4] | [v5] BottomSheetModal not usable on Android with TalkBack enabled

### ISSUE THREAD

**Issue by colintindle** [gorhom/react-native-scroll-bottom-sheet#1892]
# Bug

BottomSheetModal is not accessible with Android Talkback. Attempts to select elements using single tap inside the modal, instead focus the backdrop. Tapping the header selects elements behind/outside the modal. Swiping left/right will allow user to select any element, not just those inside the modal. 

Setting `accessible={false}` on the backdrop makes the modal body selectable with single tap, but the user can still focus on items behind the modal and backdrop. 

BottomSheetModal does not seem to present to the OS as a modal, and does not automatically take focus when opened. All of this works fine on iOS. Sadly, this issue makes the component unusable on Android for users who require the screenreader. 

## Environment info

| Library                         | Version |
| ------------------------------- | ------- |
| @gorhom/bottom-sheet            | 4.6.3   |
| react-native                    | 0.69.4  |
| react-native-reanimated         | 2.8.0   |
| react-native-gesture-handler    | 1.10.3  |

## Steps To Reproduce

1. Start Talkback
2. Open a BottomSheetModal
   - Observe that button used to open BottomSheetModal is still the active element
3. Attempt to navigate with swipe right/left
   - Observe that elements outside of modal are still focusable
   - Observe that modal contents cannot be accessed
4. Attempt to focus elements in modal header using single tap
   - Observe that elements behind the modal will be focused instead
5. Attempt to focus elements in modal body using single tap
   - Observe that BottomSheetBackdrop is focused instead

Describe what you expected to happen:

Opening the BottomSheetModal (particularly when expanded) should:
1. cause accessibility focus to be moved to the first element inside the modal
2. not allow the user to focus elements outside of the modal or backdrop

## Reproducible sample code

I have reproduced this using the example provided in the repo, on versions `v4.6.3` and `v5.0.0-alpha.10`. 

https://github.com/user-attachments/assets/7f75ac09-b5c7-4c07-a3a8-316d5b104c34

**Comment by hraschan** [comment]
I think BottomSheetBackdrop component is missing the "importantForAccessibility" prop. This is needed in order to hide the Backdrop for android. 

It would be nice if this gets implemented

**Comment by Hosam-hsm** [comment]
@colintindle Were you able to solve this issue? I am still experiencing the same issue with the latest version.

**Comment by Hosam-hsm** [comment]
@gorhom Talkback can only focus on Touchable components inside the modal and is not focusing on normal Views inside the modal in android. However, after a touchable is focused inside the modal, swiping left or right moves the focus to next or previous elements even if it is a normal View.  

I am using  @gorhom/bottom-sheet: 5.0.6 with expo: 51.0.31

**Comment by vikasmis** [comment]
It should be working by importantForAccessibility={
        isBottomSheetPresented ? 'no-hide-descendants' : 'auto'
      } on both ios and android. Make sure you pass this to top  component.   I had View >> Text>> button
----

   <View
      testID={testID}
      
      importantForAccessibility={
        isBottomSheetPresented ? 'no-hide-descendants' : 'auto'
      }
    >
      {children}. // Text and Button are here.
    </View>

//isBottomSheetPresented you can check that in bottomSheet.Length is 0 or 1

**Comment by anabeatrizzz** [comment]
Still an issue

**Comment by iaaltojarvi** [comment]
Still an issue.

**Comment by VariabileAleatoria** [comment]
> ## It should be working by importantForAccessibility={
> isBottomSheetPresented ? 'no-hide-descendants' : 'auto'
> } on both ios and android. Make sure you pass this to top component. I had View >> Text>> button
> 
> <View testID={testID}
> 
> ```
>   importantForAccessibility={
>     isBottomSheetPresented ? 'no-hide-descendants' : 'auto'
>   }
> >
>   {children}. // Text and Button are here.
> </View>
> ```
> 
> //isBottomSheetPresented you can check that in bottomSheet.Length is 0 or 1

it's not clear which component you are talking about
